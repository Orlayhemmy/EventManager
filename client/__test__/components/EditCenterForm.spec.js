import React from 'react';
import EditCenter from '../../components/CenterDetails/Template/Form/EditCenterForm';
import { centerState } from './MockData';

describe('Edit center', () => {
  const props = {
    centerState,
    showImage: jest.fn(),
    showHidden: jest.fn(),
    onChange: jest.fn()
  };

  const wrapper = shallow(<EditCenter {...props} />);
  it('should render component without errors', () => {
      expect(wrapper).toMatchSnapshot();
  });
  it('check the length of the containing div in the component', () => {
    expect(wrapper.find('#editCenterDetails').length).toEqual(1);
  });
});
