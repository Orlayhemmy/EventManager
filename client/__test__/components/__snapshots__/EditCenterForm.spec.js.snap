// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit center should render component without errors 1`] = `
<div
  hidden={true}
  id="editCenterDetails"
>
  <ImageUpload
    showImage={[MockFunction]}
  />
  <div
    className="media-body text-center mb-4"
  >
    <form
      id="edit-center-form"
    >
      <div>
        <span
          className="help-block"
        />
        <input
          className="logo text-primary text-center no-border"
          id="centerName"
          onChange={[MockFunction]}
          type="text"
          value=""
        />
        <border />
      </div>
      <h3
        className="mt-2"
      >
        location
      </h3>
      <div>
        <span
          className="help-block"
        />
        <input
          className="text-center no-border mt-0 "
          id="location"
          onChange={[MockFunction]}
          type="text"
          value=""
        />
        <border />
      </div>
      <h3
        className="mt-2"
      >
        capacity
      </h3>
      <div>
        <span
          className="help-block"
        />
        <input
          className="text-center no-border mt-0 "
          id="capacity"
          onChange={[MockFunction]}
          type="text"
          value=""
        />
        <border />
      </div>
      <h3
        className="mt-2"
      >
        facilities
      </h3>
      <p
        className="help-block"
      />
      <div>
        <input
          className="text-center no-border mt-0 "
          id="facilities"
          onChange={[MockFunction]}
          type="text"
          value=""
        />
        <border />
      </div>
      <h3
        className="mt-2"
      >
        Price
      </h3>
      <p
        className="help-block"
      />
      <div>
        <input
          className="text-center no-border mt-0 "
          id="cost"
          onChange={[MockFunction]}
          type="text"
          value=""
        />
        <border />
      </div>
      <h3
        className="mt-2"
      >
        description
      </h3>
      <p
        className="help-block"
      />
      <div>
        <textarea
          className="form-control col-xs-6 mb-4"
          id="description"
          onChange={[MockFunction]}
          value=""
        />
      </div>
      <input
        className="btn btn-sm btn-success p-1 mr-1"
        data-toggle-id="editCenterDetails"
        id="edit-submit"
        type="button"
        value="save"
      />
      <input
        className="btn btn-sm btn-danger p-1 ml-1"
        data-toggle-id="editCenterDetails"
        type="button"
        value="cancel"
      />
    </form>
  </div>
</div>
`;
